web/src/containers/activity/All.js

```diff
@@ -9,70 +9,74 @@ import {
   NameAndFundingSourceReview
 } from './NameAndFundingSource';
 import Waypoint from '../ConnectedWaypoint';
-import {
-  addActivity,
-  removeActivity,
-  updateActivity
-} from '../../actions/activities';
+import { addActivity, removeActivity } from '../../actions/editActivity';
 import { Section, Subsection } from '../../components/Section';
-import {
-  selectActivityKeys,
-  selectAllActivities
-} from '../../reducers/activities.selectors';
+import { selectAllActivities } from '../../reducers/activities.selectors';
 
-const All = ({ activityKeys, activities, add, update, remove }) => (
-  <Waypoint id="activities-overview">
-    <Section isNumbered id="activities" resource="activities">
-      <Waypoint id="activities-list" />
-      <Subsection id="activities-list" resource="activities.list" open>
-        <NameAndFundingSourceReview
-          item={activities[0]}
-          index={-1}
-          disableExpand
-        />
-        <FormAndReviewList
-          addButtonText="Add another activity"
-          allowDeleteAll
-          list={activities.slice(1)}
-          className="ds-u-border-bottom--0"
-          collapsed={NameAndFundingSourceReview}
-          expanded={NameAndFundingSourceForm}
-          noDataMessage={false}
-          onAddClick={add}
-          handleChange={update}
-          onDeleteClick={remove}
-        />
-      </Subsection>
-      {activityKeys.map((key, idx) => (
-        <Waypoint id={key} key={key}>
-          <EntryDetails aKey={key} num={idx + 1} />
-        </Waypoint>
-      ))}
-    </Section>
-  </Waypoint>
-);
+const All = ({ add, first, keys, other, remove }) => {
+  const onAdd = () => add();
+
+  const onRemove = key => {
+    keys.forEach((activityKey, i) => {
+      if (activityKey === key) {
+        remove(i);
+      }
+    });
+  };
+
+  return (
+    <Waypoint id="activities-overview">
+      <Section isNumbered id="activities" resource="activities">
+        <Waypoint id="activities-list" />
+        <Subsection id="activities-list" resource="activities.list" open>
+          <NameAndFundingSourceReview item={first} index={-1} disableExpand />
+          <FormAndReviewList
+            addButtonText="Add another activity"
+            allowDeleteAll
+            list={other}
+            className="ds-u-border-bottom--0"
+            collapsed={NameAndFundingSourceReview}
+            expanded={NameAndFundingSourceForm}
+            noDataMessage={false}
+            onAddClick={onAdd}
+            onDeleteClick={onRemove}
+          />
+        </Subsection>
+        {keys.map((key, index) => (
```

Comment:
I don't think we actually need these keys here. We have the list of activities, we can just iterate over that.